lib/storage.js

```diff
@@ -184,7 +185,19 @@ Storage.prototype.fetchTx = function(walletId, txProposalId, cb) {
   });
 };
 
+Storage.prototype.fetchTxByHash = function(hash, cb) {
+  var self = this;
+
+  this.db.collection(collections.TXS).findOne({
+    txid: hash,
+  }, function(err, result) {
+    if (err) return cb(err);
+    if (!result) return cb();
 
+    // no need to complete data for this call use cases.
```

Comment:
I suggest an alternative method `fetchTxStatusByHash` returning { id: ..., status: ... } and then based on the `status` being `accepted`, we can `fetchTx` using the `id` field.